packages/stream-api/src/streamApi.ts

```diff
@@ -62,6 +84,26 @@ export class StreamApi {
     );
   }
 
+  public async reconnect(
+    data: ReconnectPayload,
+    withResponse = true
+  ): Promise<Disposable> {
+    if (this.urlProvider == null) {
+      throw new Error('Unable to connect as no Url provider has been set');
+    }
+    await this.websocket.connect(this.urlProvider);
```

Comment:
We should try to keep the responsibility of the client simpler and have this method just be responsible for sending a reconnect message. In other words, the consumer would call:

```ts
await api.connect();
api.reconnect();
```

Understand there's a little weirdness with the naming, but think it'd be better if we keep the convention of methods that accept payloads aren't doing anything more than sending a message over the WS.